cypress/integration/filter.spec.js

```diff
@@ -16,7 +16,7 @@ describe("Reassign Tasks", () => {
   it("can hide the filters", () => {
     cy.get('#option-select-title-task-type').click()
     cy.get('#option-select-title-task-type').click()
-    cy.get('app-c-option-select__container-inner').should('not.be.visible') 
+    cy.get('.govuk-fieldset > :nth-child(1) > .govuk-checkboxes__item > .govuk-label').should('not.be.visible')
```

Comment:
In general I'd avoid using visual styling hooks for test hooks, as this makes the test brittle to change that doesn't affect whether the interaction works breaking the test. 

One option is if you have label that should not be visible, better to identify it as specifically as possible (say by the for attribute for the form element it hooks onto).

That's all assuming you aren't deliberately testing the html structure, of course.

Reply:
That's right is not testing the HTML but the behaviour. Maybe I could add a class/id with a prefix of `hook` to indicate it's only for a test?

Reply:
Potentially, but yeah if you have a label it should be for a uniquely identified element anyway (so assistive tech can associate them) which means you can us the for attribute on the label as a hook.

Looking on the Sirius dev env, I did notice an issue with those checkboxes though. HTML is:

`<input class="govuk-checkboxes__input task-type" id="Casework - Complaint review" name="selected-task-type" type="checkbox" value="CCR">
<label class="govuk-label govuk-checkboxes__label" for="">Casework - Complaint review</label>`

So it looks like id has got mixed up with a label or title attribute? For an ID I'd expect to see something like `select-task-checkbox-compliant-review` which was unique to that checkbox and was used in the label's for attribute, something like:

`  <input class="govuk-checkboxes__input task-type" **id="select-task-checkbox-compliant-review"** name="selected-task-type" type="checkbox" value="CCR">
<label class="govuk-label govuk-checkboxes__label" **for="select-task-checkbox-compliant-review"**>Casework - Complaint review</label>`

If you fixed that you'd then be able to use `label[for="select-task-checkbox-compliant-review"]` as your hook in the test.